src/components/AutocompleteTagBuilder/AutocompleteTagBuilder.int.part.tsx

```diff
@@ -0,0 +1,230 @@
+import * as React from 'react';
+import memoize from 'memoize-one';
+import { FormContextProps } from '../../hoc/withFormContext';
+import { InputChangeEvent } from '../../common';
+import { AutocompleteTagBuilderProps, AutocompleteTagBuilderState } from './AutocompleteTagBuilder.types.part';
+import { debounce } from '../../utils';
+import { TagBuilder } from '../TagBuilder';
+import { Autocomplete, AutocompleteInputProps, AutosuggestSelectEvent } from '../Autocomplete';
+
+export class AutocompleteTagBuilderInt<T> extends React.Component<
+  AutocompleteTagBuilderProps<T> & FormContextProps,
+  AutocompleteTagBuilderState<T>
+> {
+  private _fireOnInputChange: (q: string) => void;
+  private _inputNode: HTMLElement | null;
+
+  public constructor(props: AutocompleteTagBuilderProps<T>) {
+    super(props);
+
+    const { value: nullableValue, defaultValue, onInputChange, inputValue, delay = 0 } = this.props;
+    const value = nullableValue || defaultValue || [];
+    const valueMap = this.mapItemArray(value);
+
+    this.state = {
+      value: valueMap,
+      inputValue: inputValue || '',
+      controlled: props.value !== undefined || inputValue !== undefined,
+    };
+
+    this._fireOnInputChange = debounce((value: string) => {
```

Comment:
Please remove `_fireOnInputChange ` class prop and move debounce to `changeInput value`